src/batch.cpp

```diff
@@ -531,7 +531,8 @@ linear_coords_to_indices(PDE<P> const &pde, int const degree,
 // routines with explicit time advance
 template<typename P>
 void build_system_matrix(PDE<P> const &pde, elements::table const &elem_table,
-                         fk::matrix<P> &A, element_subgrid const &grid)
+                         fk::matrix<P> &A, element_subgrid const &grid,
+                         const bool use_imex, imex_flag const imex)
```

Comment:
Why do we need two `imex` flags? When we have the `explicit`, `implicit` and `unspecified` options we can use the `unspecified` option to indicate `no-imex`.

Reply:
Both flags are needed since this code was also being used for non-IMEX things - to make sure that all terms still get included when running without IMEX, even if the PDE has a term marked `imex_explicit` or `imex_implicit`. Additionally, `program_opts` was not being passed through, so this was also a quick fix without changing everything to pass `program_opts` (although the latest changes in `develop` have made `program_opts` available).

However, since the matrix-free changes have been merged into `develop` and this branch is now rebased to include those changes, this is no longer being used anymore. I would like to still keep it for now in case we need to temporarily change back to this. I can open an issue to remove this code in the future outside of this PR.

Reply:
```suggestion
                         std::optional<imex_flag> const imex)
```
Would std::optional bring us down to a single additional parameter?

Reply:
The only place where the flag is being used is in the line:
```
if (!use_imex || (use_imex && terms[k][0].flag == imex))
```

With my suggestion, the condition will become:
```
if (imex == imex::unspecified || terms[k][0].flag == imex)
```

What am I missing here?